packages/strapi-design-system/src/Json/JsonInput.js

```diff
@@ -0,0 +1,139 @@
+import React, { useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { jsonParseLinter } from '@codemirror/lang-json';
+
+import { Field, FieldLabel, FieldError, FieldHint } from '../Field';
+import { Box } from '../Box';
+import { Stack } from '../Stack';
+import { addMarks, filterMarks, lineHighlightMark } from './utils/decorationExtension';
+import { JsonInputContainer } from './JsonInputContainer';
+
+const StyledBox = styled(Box)`
+  outline: 1px solid ${({ theme, error }) => (error ? theme.colors.danger600 : 'transparent')};
+`;
+
+export const JsonInput = ({ id, label, value, error, hint, required, theme, onChange, disabled, labelAction }) => {
+  const [errorMessage, setErrorMessage] = useState(error);
```

Comment:
ℹ️  Not sure about experience in the CMS yet but I thought it made sense to get more feedback about the error than just the highlighting.

Reply:
Decided against this for now and instead just made the component behave as close to the current component as possible...